nad_ch/controllers/web/routes.py

```diff
@@ -1,4 +1,8 @@
 from flask import Blueprint, current_app, render_template, g
```

Comment:
Some tests for the routes would be nice to have: https://flask.palletsprojects.com/en/3.0.x/testing/

Given how admirably humble they are, I guess they'd mostly just act like smoke tests. Maybe end-to-end testing would be preferable for that purpose?

Reply:
I agree some sort of test to ensure the routes are wired up correctly would be useful, especially as we build out the web UI. I added a couple that exposed some gaps in how the controllers were handling the output of the use case.